src/pointer-analysis/value_set.cpp

```diff
@@ -344,6 +344,30 @@ void value_sett::get_value_set(
   get_value_set_rec(tmp, dest, "", tmp.type(), ns);
 }
 
+/// Check if 'suffix' starts with 'field'.
+/// Suffix is delimited by periods and '[]' (array access) tokens, so we're
+/// looking for ".field($|[]|.)"
+static bool suffix_starts_with_field(
+  const std::string &suffix, const std::string &field)
```

Comment:
I think each call of this function would like to pass an `irep_idt` as second argument; I'd thus suggest to do the `id2string` within this function. The same is true for `strip_first_field_from_suffix`.